tic-tac-toe/src/tic_tac_toe/board.clj

```diff
@@ -0,0 +1,79 @@
+(ns tic-tac-toe.board
+  (:require [tic-tac-toe.marks :refer :all]))
+
+(def dimension 3)
+
+(defn- one-based [value]
+  (+ value 1))
+
+(defn- three-matching-player-symbols? [row]
+  (let [[one two three] row]
+    (and (= one two three) (not(= one nil)))))
+
+(defn- winning-row? [rows]
+  (some true? (map
+                (fn[row] (three-matching-player-symbols? row))
+                rows )))
+
+(defn- find-winning-symbol [rows]
+  (first
+    (filter (fn[matching-symbols] (not (= matching-symbols nil)))
+            (map (fn[row] (if (three-matching-player-symbols? row) (first row) nil)) rows))))
+
+(defn create [board-config]
+  board-config)
+
+(defn place-mark[board mark index]
+  (assoc board index mark))
+
+(defn get-mark-at-index [board index]
+  (get board index))
+
+(defn get-rows[board]
+  [
+   (subvec board 0 dimension)
+   (subvec board dimension (* 2 dimension))
+   (subvec board (* 2 dimension) (* 3 dimension))
+   ])
+
+(defn get-columns[board]
+  (get-rows (vec (flatten (conj (map (fn[row] (first row)) (get-rows board))
+                                (map (fn[row] (second row) ) (get-rows board))
+                                (map (fn[row] (nth row 2)) (get-rows board)))))))
+(defn get-diagonals[board]
+  [
+   (vector
+     (get-mark-at-index board 0)
+     (get-mark-at-index board 4)
+     (get-mark-at-index board 8))
+   (vector
+     (get-mark-at-index board 2)
+     (get-mark-at-index board 4)
+     (get-mark-at-index board 6))
+   ]
+  )
+
+(defn winning-line? [board]
+  (if (= true (or
```

Comment:
Ah I see. One option is to replace `if` statement with a call to `boolean` in that case.

https://clojuredocs.org/clojure.core/boolean